src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/CodeGen/ReadyToRunObjectWriter.cs

```diff
@@ -79,10 +81,39 @@ public void EmitPortableExecutable()
                 if (mapFile != null)
                     mapFile.WriteLine($@"R2R object emission started: {DateTime.Now}");
 
+                PEHeaderBuilder headerBuilder;
+                int timeDateStamp;
+                ISymbolNode r2rHeaderExportSymbol;
+
+                if (_nodeFactory.CompilationModuleGroup.IsCompositeBuildMode && _componentModule == null)
+                {
+                    headerBuilder = PEHeaderProvider.Create(
+                        characteristics: Characteristics.ExecutableImage | Characteristics.Dll,
+                        dllCharacteristics: default(DllCharacteristics),
+                        Subsystem.Unknown,
+                        _nodeFactory.Target);
+                    timeDateStamp = 0;
+                    r2rHeaderExportSymbol = _nodeFactory.Header;
```

Comment:
We should never force the PE timestamp to zero. That will cause problems with getting a pdb to work. Instead, we should either copy a timestamp, as we do already for the non-composite scenario, or use a deterministic id provider. (Basically, I'm arguing that this timestamp parameter should be passed as a nullable int, and this case should be null.

Reply:
And not write a new timestamp. 

Or, we should implement a deterministic id provider, and capture the timestamp out of that. 

The general path for that is:
Func<IEnumerable<Blob>, BlobContentId> deterministicProvider = new new Func<IEnumerable<Blob>, BlobContentId>(content => BlobContentId.FromHash(CryptographicHashProvider.ComputeHash(HashAlgorithmName.SHA1, content));

then pass that provider to the PEBuilder constructor. See PeWriter.cs in the Roslyn codebase for details, or ask the Roslyn folks for advise on how to do that.

---
In reply to: [379591234](https://github.com/dotnet/runtime/pull/31663#discussion_r379591234) [](ancestors = 379591234)

Reply:
I have created the following tracking issue for this: https://github.com/dotnet/runtime/issues/32507